lib/rules/hashbang.js

```diff
@@ -12,10 +12,50 @@ const { getPackageJson } = require("../util/get-package-json")
 const getNpmignore = require("../util/get-npmignore")
 const { isBinFile } = require("../util/is-bin-file")
 
-const NODE_SHEBANG = "#!/usr/bin/env node\n"
+const ENV_SHEBANG = "#!/usr/bin/env"
+const NODE_SHEBANG = `${ENV_SHEBANG} {{executableName}}\n`
 const SHEBANG_PATTERN = /^(#!.+?)?(\r)?\n/u
-const NODE_SHEBANG_PATTERN =
-    /^#!\/usr\/bin\/env(?: -\S+)*(?: [^\s=-]+=\S+)* node(?: [^\r\n]+?)?\n/u
+
+// -i -S
+// -u name
+// --ignore-environment
+// --block-signal=SIGINT
+const ENV_FLAGS = /^\s*-(-.*?\b|[ivS]+|[Pu](\s+|=)\S+)(?=[ \b])/
+
+// NAME="some variable"
+// FOO=bar
+const ENV_VARS = /^\s*\w+=(?:"(?:[^"\\]|\\.)*"|\w+)/
+
+/**
+ * @param {string} shebang
+ * @param {string} executableName
+ * @returns {boolean}
+ */
+function isNodeShebang(shebang, executableName) {
+    if (shebang == null || shebang.length === 0) {
+        return false
+    }
+
+    shebang = shebang.slice(shebang.indexOf(ENV_SHEBANG) + ENV_SHEBANG.length)
+    while (ENV_FLAGS.test(shebang) || ENV_VARS.test(shebang)) {
```

Comment:
can we use a `/g` flag to replace `while`?

Reply:
you don't have to write to one regex:
```js
shebang = shebang.replace(/.../g, "").replace(/.../g, "")
```

Reply:
You can see that here: (I added the `m` flag to make it visible)

<img width="500" alt="image" src="https://github.com/eslint-community/eslint-plugin-n/assets/2230835/90e2b149-ac70-4663-9e46-9578f5c16f1a">

Reply:
I see, missed it somehow. 😅

Reply:
Its because regex is evil to debug! 🤣